src/state/middlewares/assetsMiddleware.ts

```diff
@@ -0,0 +1,34 @@
+import { AssetService } from '@shapeshiftoss/asset-service'
+import { Asset, ChainTypes, NetworkTypes } from '@shapeshiftoss/types'
+import { assets } from 'state/slices/assetsSlice/assetsSlice'
+
+export const assetMiddleware = (store: any) => {
+  const service = new AssetService('')
+
+  const fetchAsset = async ({
+    tokenId,
+    chain,
+    network
+  }: {
+    tokenId?: string
+    chain: ChainTypes
+    network: NetworkTypes
+  }) => {
+    if (!service.isInitialized) await service.initialize()
```

Comment:
I like the option of easily being able to pull both, but that's a good point, it could get bad.  We'd have to be on that when doing code reviews.  We could also let the dev know in the description of the reducer on what situations to use what selector.  My vote is for the first solution